x/bank/keeper/keeper.go

```diff
@@ -430,6 +430,7 @@ func (k BaseKeeper) trackDelegation(ctx sdk.Context, addr sdk.AccAddress, balanc
 	if ok {
 		// TODO: return error on account.TrackDelegation
 		vacc.TrackDelegation(ctx.BlockHeader().Time, balance, amt)
+		k.ak.SetAccount(ctx, acc)
```

Comment:
I'm a little unlcear how type assertions work - is `vacc` a copy of `acc`? If so, I believe you want
```suggestion
		k.ak.SetAccount(ctx, vacc)
```

As the call to `TrackUndelegation` alters the state of `vacc` not `acc`. 

Same would be true on L449.

Reply:
It's the same underlying account in memory, so I do not believe it matters.

In fact, this is more or elss the same as the original code: https://github.com/cosmos/cosmos-sdk/blob/v0.39.2/x/bank/internal/keeper/keeper.go#L78-L82

Reply:
Correct.